builtin/logical/database/path_config_connection.go

```diff
@@ -613,6 +564,92 @@ func storeConfig(ctx context.Context, storage logical.Storage, name string, conf
 	return nil
 }
 
+func (b *databaseBackend) getPinnedVersion(ctx context.Context, pluginName string) (string, error) {
+	extendedSys, ok := b.System().(logical.ExtendedSystemView)
+	if !ok {
+		return "", fmt.Errorf("database backend does not support running as an external plugin")
```

Comment:
This was already true because the database plugin uses `logical.SystemView` methods that are not available over the gRPC implementation of SystemView. In hindsight, those methods (like ListVersionedPlugins) probably should have been added to `logical.ExtendedSystemView` as well.